alkira/resource_alkira_service_pan.go

```diff
@@ -389,11 +416,47 @@ func generateServicePanRequest(d *schema.ResourceData, m interface{}) (*alkira.S
 	panoramaIpAddresses := convertTypeListToStringList(d.Get("panorama_ip_addresses").([]interface{}))
 	panoramaTemplate := d.Get("panorama_template").(string)
 
+	panCredName := d.Get("name").(string) + randomNameSuffix()
+	panCredential := alkira.CredentialPan{
+		Username: d.Get("username").(string),
+		Password: d.Get("password").(string),
+	}
+
+	panCredentialId, err := client.CreateCredential(panCredName, alkira.CredentialTypePan, panCredential, 0)
+
+	if err != nil {
+		log.Printf("[ERROR] failed to process PAN firewall credentials, %v", err)
+		return nil, err
+	}
+
 	instances, err := expandPanInstances(d.Get("instance").([]interface{}), m)
 	if err != nil {
 		return nil, err
 	}
 
+	for i, instance := range instances {
+		panInstanceCredential := alkira.CredentialPanInstance{
+			AuthCode: instance.AuthCode,
+			AuthKey:  instance.AuthKey,
+		}
+		panInstanceExpiry, err := convertInputTimeToEpoch(instance.Expiry)
+		if err != nil {
+			log.Printf("[ERROR] failed to parse 'pan_instance_exiry', %v", err)
+			return nil, err
+		}
+		panInstanceId, err := client.CreateCredential(
```

Comment:
Same thing here. This will create a new credential every single time. That is the behavior that we want on CREATE not UPDATE.